### ISSUE TITLE
GGMAX DEVTEST - Zones Not Working Correctly

### ISSUE THREAD

**Issue by OldFlak** [TheGameCreators/GameGuruRepo#5691]
Changing the boundary points of zones makes them inaccurate.

When a zone has a large coverage area, the zone only triggers when you are near the center of it.

They are also easily crash MAX to desktop when fiddling with them.

**Comment by Necrym59** [comment]
Confirmed.
Maybe a sprayed zone maybe better concept for future

**Comment by AmenMoses** [comment]
Zones should simply be 3d or 2d shapes.
They should be nameable and be connectable with the blobby logic doodahs etc.
For simplicity they could probably just be an adjustable polygon in x/z plane with an adjustable height in y (i.e. thickness).
Lua commands similar to my (https://forum.game-guru.com/thread/218470#msg2584479) GGC zonelib should then be provided that allow simple checks for player in named zone(s), player entering/leaving zones, entity versions of the same etc.

There is a super efficient point in poly function in my zonelib btw, would be easy enough to convert to C++ or just keep it all in Lua and use the old waypoint system or something similar to store the zones under the covers.

**Comment by OldFlak** [comment]
**Zones should simply be 3d or 2d shapes.** 

That would be great - the current zones are and have always been finicky at best. They are really only good if left stock size and shape when using them.

**Comment by LeeBamberTGC** [comment]
@OldFlak I made a MASSIVE zone and triggered a sound to play when the player crosses its border, and it worked from all edges, even the concave edge. Can you try with the latest version and if exists, attach the stock FPM, thanks:
![image](https://github.com/user-attachments/assets/92fdafc6-2ec3-4ec3-b39a-4e6a57c51246)

**Comment by Kasseyus** [comment]
@LeeBamberTGC Verified

Sample Level Image

![Trigger Zone NPC](https://github.com/user-attachments/assets/3a8f7129-0505-454c-9d48-933352f5c29d)

**Comment by OldFlak** [comment]
Still completely busted for me Lee
I must add that - to be frank - zones are still simply terrible to work with (sometimes impossible) when trying to place them near, or worse still, on top of models (platforms up in the air for example). we really need something better.

I really don't think this is a medium issue 
- it makes zones all but useless, unless they are on the ground, and you don't change the shape.

In short zones are just not a quality component of MAX - he really deserves better.

Thanks for considering :)

**Comment by LeeBamberTGC** [comment]
@Kasseyus Are you saying that when you set the zone marked in yellow to something like play sound, it does not trigger when you cross the zone lines?
![image](https://github.com/user-attachments/assets/dfedafb5-a464-47fe-ab93-a51d0a708bc7)

**Comment by Kasseyus** [comment]
@LeeBamberTGC

I think I mixed up the images here.  This issue should have had the following image:
![image](https://github.com/user-attachments/assets/3cc5655d-65ca-4cc2-880f-2b8113a08056)

In this case, the trigger only triggered once you got close to the central node, rather than passing any part

 In this particular case, the NPC will walk back and forth over the trigger zone, but will not TRIGGER the zone until the player in within the 'empty' terrain texture from where they start.  This image should have gone with https://github.com/TheGameCreators/GameGuruRepo/issues/5753

**Comment by Necrym59** [comment]
There are going to be times when zones need to be always active regardless of the players distance can we not get a ZoneAlwaysActive() lua comnmand or somthing that would solve all this.

**Comment by LeeBamberTGC** [comment]
@Necrym59 That is a good point, it could VERY WELL be that the zone is not detecting the player at the boundary as the player was too far away from the node source of the zone, I will check that first :)

**Comment by LeeBamberTGC** [comment]
@Kasseyus No joy, can you send me your simple level so I can cross the barrier of the zone and not trigger. Here is mine and the sound triggers the moment I cross the near barrier:
![image](https://github.com/user-attachments/assets/739dc29c-9f7d-474a-8e45-1b25c914853c)

**Comment by plemsoft** [comment]
@LeeBamberTGC I fixed the above here: https://github.com/TheGameCreators/GameGuruRepo/issues/5753 , they are now running every half sec. when at those distances. I did not see that it was also reported here, so only marked the other issue :)

**Comment by LeeBamberTGC** [comment]
@plemsoft I am assuming I've already compiled the latest commits from you related to this issue, so fingers crossed they are now part of DEVTEST :)

@OldFlak Can you check the latest DEVTEST and confirm the zones as described above now work for you?  If so, we can close this issue.

**Comment by OldFlak** [comment]
@LeeBamberTGC @plemsoft 
Yes, the initial issue for this thread is fixed in current DEVTEST 👍 

I will make separate post with regard to Zones being unfriendly to work with.

Thanks for sorting :)
